vmm/src/device_manager.rs

```diff
@@ -1347,11 +1363,21 @@ impl DeviceManager {
                 vgic_config.msi_addr + vgic_config.msi_size - 1,
             )
         }
+        #[cfg(target_arch = "riscv64")]
+        {
+            let vcpus = self.config.lock().unwrap().cpus.boot_vcpus;
+            let vaia_config = aia::Aia::create_default_config(vcpus.into());
+            (
+                vaia_config.imsic_addr,
+                vaia_config.imsic_addr + vaia_config.vcpu_count as u64 * arch::layout::IMSIC_SIZE
+                    - 1,
+            )
+        }
         #[cfg(target_arch = "x86_64")]
         (0xfee0_0000, 0xfeef_ffff)
```

Comment:
Future @rbradford TODO: Let's refactor this so we have a different version of the whole function per architecture - would remove some of the rather annoying indentation.